test/utils/Base64.test.js

```diff
@@ -43,11 +44,30 @@ describe('Strings', function () {
     ])
       it(title, async function () {
         const buffer = Buffer.from(input, 'ascii');
-        expect(await this.mock.$encodeURL(buffer)).to.equal(base64toBase64Url(ethers.encodeBase64(buffer)));
-        expect(await this.mock.$encodeURL(buffer)).to.equal(expected);
+        await expect(this.mock.$encodeURL(buffer)).to.eventually.equal(base64toBase64Url(ethers.encodeBase64(buffer)));
+        await expect(this.mock.$encodeURL(buffer)).to.eventually.equal(expected);
+        await expect(this.mock.$decode(expected)).to.eventually.equal(ethers.hexlify(buffer));
       });
   });
 
+  it('Decode invalid base64 string', async function () {
+    const getHexCode = str => ethers.hexlify(ethers.toUtf8Bytes(str));
+    const helper = { interface: ethers.Interface.from(['error InvalidBase64Digit(bytes1)']) };
+
+    // ord('$') < 43
+    await expect(this.mock.$decode('dGVzd$=='))
+      .to.be.revertedWithCustomError(helper, 'InvalidBase64Digit')
+      .withArgs(getHexCode('$'));
+    // ord('~') > 122
+    await expect(this.mock.$decode('dGVzd~=='))
+      .to.be.revertedWithCustomError(helper, 'InvalidBase64Digit')
+      .withArgs(getHexCode('~'));
```

Comment:
Very minor, maybe that to check range edges:
```suggestion
    // ord('*') < 43
    await expect(this.mock.$decode('dGVzd*=='))
      .to.be.revertedWithCustomError(helper, 'InvalidBase64Digit')
      .withArgs(getHexCode('*'));
    // ord('{') > 122
    await expect(this.mock.$decode('dGVzd{=='))
      .to.be.revertedWithCustomError(helper, 'InvalidBase64Digit')
      .withArgs(getHexCode('{'));
```